### PR TITLE
Fallback to userhome when no "file" scheme folders open

### PR DESCRIPTION
**Pull request by alexr00**
Fixes https://github.com/github/codespaces/issues/1695

### CODE REVIEW

**Review by bpasero — COMMENTED** [review]
I am not entirely sure, should this maybe go into the implementation of `defaultFolderPath` and/or `defaultFilePath` in case others hit this?

**Comment by isidorn** [comment]
@alexr00 thanks a lot for looking into this!
This is the in our remote release actual issue https://github.com/microsoft/vscode-remote-release/issues/2536
So when doing something here we should try to fix the other one as well. 

Makes sense to put this in the `defaultFolderPath` and `defaultFilePath` as @bpasero suggests

**Comment by alexr00** [comment]
Done.

File: src/vs/workbench/contrib/notebook/browser/notebookEditorInput.ts [hunk 1]
```diff
@@ -145,7 +143,7 @@ ${patterns}
 	}
 
 	async suggestName(suggestedFilename: string) {
-		return joinPath(this._fileDialogService.defaultFilePath() || (await this._pathService.userHome()), suggestedFilename);
+		return joinPath(await this._fileDialogService.defaultFilePath(), suggestedFilename);
```
Comment (alexr00) [inline src/vs/workbench/contrib/notebook/browser/notebookEditorInput.ts:146]: @rebornix FYI

File: src/vs/workbench/contrib/searchEditor/browser/searchEditorInput.ts [hunk 2]
```diff
@@ -276,7 +276,7 @@ export class SearchEditorInput extends EditorInput {
 
 		const searchFileName = (query.replace(/[^\w \-_]+/g, '_') || 'Search') + SEARCH_EDITOR_EXT;
 
-		return joinPath(this.fileDialogService.defaultFilePath(this.pathService.defaultUriScheme) || (await this.pathService.userHome()), searchFileName);
+		return joinPath(await this.fileDialogService.defaultFilePath(this.pathService.defaultUriScheme), searchFileName);
```
Comment (alexr00) [inline src/vs/workbench/contrib/searchEditor/browser/searchEditorInput.ts:279]: @JacksonKearl FYI

**Review by bpasero — APPROVED** [review]
Looks great and seems to fix https://github.com/microsoft/vscode-remote-release/issues/2536 as well actually 👍

**Comment by isidorn** [comment]
Great work 👏

**Comment by BIGBALLON** [comment]
Great work

**Pull request merged**